Atomic/syscontainers.py

```diff
@@ -1694,15 +1694,26 @@ def get_system_images(self, get_all=False, repo=None):
         :returns: A list of image data structures.
         :rtype: list
         """
+        def get_img_ref_str(ostree_ref_str):
+            return ostree_ref_str.replace(OSTREE_OCIIMAGE_PREFIX, "")
+
         if repo is None:
             repo = self._get_ostree_repo()
             if repo is None:
                 return []
         revs = [x for x in repo.list_refs()[1] if x.startswith(OSTREE_OCIIMAGE_PREFIX) \
-                and (get_all or len(x) != len(OSTREE_OCIIMAGE_PREFIX) + 64)]
```

Comment:
I think it is better if we leave also the condition on the length.  Otherwise we will catch images like `cafe`, `c0`..

Reply:
Hmm, Looking more into the code. Looks like we encode the ref via the following function.
`_encode_to_ostree_ref`
```
 def convert(x):
            return (x if str.isalnum(str(x)) or x in '.-' else "_%02X" % ord(x))

        if name.startswith("oci:"):
            name = name[len("oci:"):]
        registry, image, tag = SystemContainers._parse_imagename(name)
        if registry:
            fullname = "%s/%s:%s" % (registry, image, tag)
        else:
            fullname = "%s:%s" % (image, tag)

        ret = "".join([convert(i) for i in fullname])
        return ret
```

and in parse_image_name
```
 sep = image.find(":")
        if sep > 0:
            return reg, image[:sep], image[sep + 1:]
        else:
            return reg, image, "latest"
```

After digging a bit more, we seem always include a ":" for tagging when we encode the ref, no matter what the tag is. (i.e: either latest or other). The encoded ref itself will then always have a ":" which get encoded to `_3A` in unicode. In this case, I still suggest we drop the check of the length for encoded ref here?

But I am also fine with adding it back in this place, if you feel strongly about this =P.

Reply:
I'll defer to @giuseppe.